### PR TITLE
added few collections utils

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/flanksource/commons
 
-go 1.17
+go 1.20
```
Comment (adityathebe) [inline go.mod:3]: updated go version since this PR makes use of Go generics

Comment (yashmehrotra) [reply go.mod:3]: Also update the go version in all the `.github/workflow` files for lint etc.

Comment (yashmehrotra) [reply go.mod:3]: I see that this doesn't have a lint check, can you copy the one from other repos here

Comment (adityathebe) [reply go.mod:3]: sure

File: certs/cert.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func DecryptCertificate(cert []byte, privateKey []byte, password []byte) (*Certi
 	block, _ := pem.Decode(privateKey)
 
 	var decrypted []byte
-	if decrypted, err = x509.DecryptPEMBlock(block, password); err != nil {
+	if decrypted, err = x509.DecryptPEMBlock(block, password); err != nil { //nolint:staticcheck
```
Comment (adityathebe) [inline certs/cert.go:40]: `x509.DecryptPEMBlock` is deprecated. I'm just suppressing the linter error for now.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -11,9 +11,9 @@ jobs:
       matrix:
         platform: [ubuntu-latest]
         go-version:
+          - 1.20.x
           - 1.19.x
```
Comment (adityathebe) [inline .github/workflows/test.yml:15]: Do we need to test in all of these versions?

**Comment by adityathebe** [comment]
I had to fix a couple of linter errors. Some `errCheck` errors are suppressed because the func did not return any error. Didn't want to change the func def. Maybe I could just log those errors.

File: .github/workflows/test.yml (hunk 3 above)
Comment (yashmehrotra) [reply .github/workflows/test.yml:15]: No, lets remove non go-generics supported ones

File: certs/cert.go (hunk 2 above)
Comment (yashmehrotra) [reply certs/cert.go:40]: Cool

Comment (yashmehrotra) [reply certs/cert.go:40]: Add this as a comment in the code too, so that we'll know the reason in the future

**Review by yashmehrotra — APPROVED** [review]

**Pull request merged**